v8_test.go

```diff
@@ -963,6 +963,52 @@ func TestTypedArrayBuffers(t *testing.T) {
 	}
 }
 
+func TestGetPromiseResult(t *testing.T) {
+	t.Parallel()
+	ctx := NewIsolate().NewContext()
+
+	// Resolved promise
+	prom, err := ctx.Eval(`new Promise((resolve, reject)=>{
+		resolve("hello")
+	})`, "test.js")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	res, err := prom.GetPromiseResult()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	str := res.String()
+	if str != "hello" {
+		t.Errorf("Expected promise to resolve to 'hello', but got: %s", str)
+	}
+
+	// Rejected promise
+	prom, err = ctx.Eval(`new Promise((resolve, reject)=>{
+		reject(new Error("hello"))
+	})`, "test.js")
+
+	res, err = prom.GetPromiseResult()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	str = res.String()
+	if str != "Error: hello" {
```

Comment:
This is not great.  I don't want an API where you have to detect whether the returned value is an Error or not.  It seems that this is pointing, again, towards having multiple types of which one is v8.Error that _also_ implements the _error_ interface.

There needs to be a better way of determining whether it has been rejected or not.

What if GetPromiseResult returns `(State, *Value, error)`?

If State == rejected, the Error object probably has a backtrace and stuff.

Reply:
Probably going to wait until we figure out the whole "Kind" thing before I get back to this one.